File: backend/src/models/MenuItem.js
const mongoose = require('mongoose');

const toppingSchema = new mongoose.Schema({
  name: {
    type: String,
    required: true,
    trim: true
  },
  extraPrice: {
    type: Number,
    default: 0,
    min: 0
  },
  isDefault: {
    type: Boolean,
    default: false
  },
  isAvailable: {
    type: Boolean,
    default: true
  },
  sortOrder: {
    type: Number,
    default: 0
  }
}, { _id: true });

const toppingGroupSchema = new mongoose.Schema({
  name: {
    type: String,
    required: true,
    trim: true
  },
  isRequired: {
    type: Boolean,
    default: false
  },
  minSelect: {
    type: Number,
    default: 0,
    min: 0
  },
  maxSelect: {
    type: Number,
    default: 1,
    min: 1
  },
  sortOrder: {
    type: Number,
    default: 0
  },
  toppings: [toppingSchema]
}, { _id: true });

const menuItemSchema = new mongoose.Schema({
  category: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'Category',
    required: true
  },
  name: {
    type: String,
    required: true,
    trim: true,
    maxlength: 200
  },
  description: String,
  imageUrl: String,
  price: {
    type: Number,
    required: true,
    min: 0
  },
  costPrice: {
    type: Number,
    default: 0,
    min: 0
  },
  unit: {
    type: String,
    default: 'phần',
    trim: true
  },
  status: {
    type: String,
    enum: ['available', 'out_of_stock', 'suspended'],
    default: 'available'
  },
  isPopular: {
    type: Boolean,
    default: false
  },
  isNew: {
    type: Boolean,
    default: false
  },
  preparationTime: {
    type: Number, // minutes
    default: 15
  },
  sortOrder: {
    type: Number,
    default: 0
  },
  toppingGroups: [toppingGroupSchema]
}, {
  timestamps: true
});

menuItemSchema.index({ category: 1 });
menuItemSchema.index({ status: 1 });
menuItemSchema.index({ isPopular: 1 });
menuItemSchema.index({ isNew: 1 });
menuItemSchema.index({ name: 'text', description: 'text' });

module.exports = mongoose.model('MenuItem', menuItemSchema);

File: mobile/core/domain/src/main/java/com/qrdatmon/core/domain/model/MenuItem.kt
package com.qrdatmon.core.domain.model

data class MenuItem(
    val id: String,
    val categoryId: String,
    val name: String,
    val description: String?,
    val imageUrl: String?,
    val price: Double,
    val unit: String = "phần",
    val status: String = "available",
    val isPopular: Boolean = false,
    val isNew: Boolean = false,
    val preparationTime: Int = 15,
    val toppingGroups: List<ToppingGroup> = emptyList()
)

data class ToppingGroup(
    val id: String,
    val name: String,
    val isRequired: Boolean = false,
    val minSelect: Int = 0,
    val maxSelect: Int = 1,
    val toppings: List<Topping> = emptyList()
)

data class Topping(
    val id: String,
    val name: String,
    val extraPrice: Double = 0.0,
    val isDefault: Boolean = false,
    val isAvailable: Boolean = true
)
